File: internal/services/tour_service.go
package services

import (
	"tourist-app/internal/models"
	"tourist-app/internal/repository"
)

type TourService struct {
	repo *repository.TourRepository
}

func NewTourService(repo *repository.TourRepository) *TourService {
	return &TourService{repo: repo}
}

func (s *TourService) GetAllTours() ([]models.Tour, error) {
	return s.repo.GetAll()
}

func (s *TourService) GetTourByID(id int) (*models.Tour, error) {
	return s.repo.GetByID(id)
}

func (s *TourService) CreateTour(tour *models.Tour) error {
	return s.repo.Create(tour)
}

func (s *TourService) UpdateTour(tour *models.Tour) error {
	return s.repo.Update(tour)
}

func (s *TourService) DeleteTour(id int) error {
	return s.repo.Delete(id)
}

File: tourist-app-frontend/src/services/tourService.js
import api from './api';

export const getTours = () => api.get('/tours');
export const getTour = (id) => api.get(`/tours/${id}`);
export const createTour = (tourData) => api.post('/tours', tourData);
export const updateTour = (id, tourData) => api.put(`/tours/${id}`, tourData);
export const deleteTour = (id) => api.delete(`/tours/${id}`);
